src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.cpp

```diff
@@ -126,263 +157,211 @@ void KernelEmitter::validate_arguments(const std::vector<size_t> &in,
-    //  we need a more elegant approach to avoid a full copy here
-    auto local_gpr_pool = gp_regs_pool;
-    local_gpr_pool.push_back(static_cast<size_t>(reg_indexes.getIdx()));
-    local_gpr_pool.push_back(static_cast<size_t>(reg_const_params.getIdx()));
     for (const auto& c : body) {
         const auto& emitter = c.first;
         std::vector<size_t> in_regs, out_regs;
         std::tie(in_regs, out_regs) = c.second;
-        if (auto tile_scheduler = std::dynamic_pointer_cast<TileSchedulerEmitter>(emitter))
-            out_regs = gp_regs_used;
-        emitter->emit_code(in_regs, out_regs, vec_regs_pool, local_gpr_pool);
+        emitter->emit_code(in_regs, out_regs, vec_regs_pool, gp_regs_pool);
     }
     h->postamble();
 }
 
-TileSchedulerEmitter::TileSchedulerEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
-                                           const std::shared_ptr<ov::Node>& n) : jit_container_emitter(h, isa, n) {
-    const auto tile_scheduler = ov::as_type_ptr<ngraph::snippets::op::TileScheduler>(n);
-    if (!tile_scheduler)
-        IE_THROW() << "TileSchedulerEmitter invoked with invalid op argument";
-    if (!tile_scheduler->compile_params)
-        IE_THROW() << "TileEmitter invoked without compile_params";
-    body = {tile_scheduler->vector_region, tile_scheduler->scalar_region};
-    jcp = *reinterpret_cast<const jit_snippets_compile_args*>(tile_scheduler->compile_params);
-}
-void TileSchedulerEmitter::emit_code(const std::vector<size_t> &in,
-                                     const std::vector<size_t> &out,
-                                     const std::vector<size_t> &pool,
-                                     const std::vector<size_t> &gpr) const {
+
+LoopBeginEmitter::LoopBeginEmitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                         const std::shared_ptr<ov::Node>& n) : jit_emitter(h, isa, n) {
+    loop_begin = ov::as_type_ptr<ngraph::snippets::op::LoopBegin>(n);
+    if (!loop_begin)
+        IE_THROW() << "LoopBeginEmitter invoked with invalid op argument";
+    const auto& target_inputs = loop_begin->output(loop_begin->get_output_size() - 1).get_target_inputs();
+    // todo: this check could be excessive, since we check for it in validate_and_infer_types()
+    if (target_inputs.size() != 1)
+        IE_THROW() << "LoopBeginEmitter invoked with invalid configuration: the last output must have exactly one input attached";
+    const auto loop_end = ov::as_type_ptr<ngraph::snippets::op::LoopEnd>(target_inputs.begin()->get_node()->shared_from_this());
+    if (!loop_end)
+        IE_THROW() << "LoopBeginEmitter invoked with invalid configuration: the last output must be LoopEnd";
+    work_amount = loop_begin->get_work_amount();
+    evaluate_once = loop_begin->get_evaluate_once();
+    num_inputs = loop_begin->get_output_size() - 1;
```

Comment:
Could you please clarify why do you use `get_output_size() - 1` instead of `get_input_size()` to get `input_count`? It's not a clear..

Reply:
For LoopBegin, `get_input_size()`  is always  equal to `get_output_size() - 1` [by construction](https://github.com/IvanNovoselov/openvino/blob/4497d48fa01c045474277006241ed99db0ed8ede/src/common/snippets/src/op/loop.cpp#L59-L60), so they are interchangeable.
I see your point however, let's use `get_inputs_size()` for better readability.